source/utilities.cc

```diff
@@ -2507,25 +2525,25 @@ namespace aspect
 
 
 
-    std::vector<Operator> create_model_operator_list(const std::vector<std::string> &operator_names)
+    std::vector<Operator>
+    create_model_operator_list(const std::vector<std::string> &operator_names,
+                               const std::string &id_text)
     {
       std::vector<Operator> operator_list(operator_names.size());
       for (unsigned int i=0; i<operator_names.size(); ++i)
         {
-          // create operator list
-          if (operator_names[i] == "add")
-            operator_list[i] = Operator(Operator::add);
-          else if (operator_names[i] == "subtract")
-            operator_list[i] = Operator(Operator::subtract);
-          else if (operator_names[i] == "minimum")
-            operator_list[i] = Operator(Operator::minimum);
-          else if (operator_names[i] == "maximum")
-            operator_list[i] = Operator(Operator::maximum);
-          else
-            AssertThrow(false,
-                        ExcMessage ("ASPECT only accepts the following operators: "
-                                    "add, subtract, minimum and maximum. But your parameter file "
-                                    "contains: " + operator_names[i] + ". Please check your parameter file.") );
+          try
+            {
+              operator_list[i] = Operator(operator_names[i]);
+            }
+          catch (Operator::ExcOperatorUnknown &exc)
+            {
+              AssertThrow(false,
+                          ExcMessage ("ASPECT only accepts the following operators: "
+                                      "add, subtract, minimum and maximum. But your parameter file "
```

Comment:
Is there a way to get the operators we allow from the Operator object instead of spelling them out by hand? This would allow adding operators in the future without having to change the error message by hand (which we might forget).